Rebuild and merge context menu code

winston-de opened the pull request:
**Resolved / Related Issues**

Itemize resolved / related issues by this PR.
- Closes #2747
 
**Details of Changes**
This is still a work in progress.

#### Goals:
- Separate the context menu items from the MenuFlyout control (and UI in general) to make it easier to implement #2342 (and possibly #2780) in the future.
- Add code to allow for customization of items to make it easier to implement #3966


## ToDo:
- [x] Remove the ".Text" from the string resource keys, and use the localized strings
- [x] Add recycle bin items
- [x] Add the layout flyout
- [x] Clean up/remove old context menu code in BaseLayout
- [ ] Add pin to start commands (depends on #4218)
- [x] Implement IsChecked for the model
- [x] Move items that are only shown on shift to the more menu if shift isn't held 
- [x] Closes #4331 

**Validation**

How did you test these changes?
- [x] Built and ran the app
- [x] Tested the changes for accessibility

**Screenshots (optional)**

Add screenshots here.

winston-de commented:
Alright, I got most of the basic menu items implemented.

Without & with shift held.
![image](https://user-images.githubusercontent.com/59544401/112557426-ab5d8180-8d89-11eb-8ec5-db65d9105787.png)

yaichenbaum commented:
@winston-de What still needs to be done?

winston-de commented:
@yaichenbaum I still have to add support custom font families for the glyphs (which is why the shortcut icon is wrong).
Also
- [ ] Remove the ".Text" from the string resource keys
- [ ] Add recycle bin items
- [ ] Add the base layout flyout
- [ ] Clean up/remove old context menu code in BaseLayout

yaichenbaum commented:
@winston-de Sounds like good progress, this will make maintaining the context menu so much easier!

winston-de commented:
@yaichenbaum I've updated all the localization keys (removed the ".Text"). Should I run the multilingual app toolkit?

winston-de commented:
I don't know what do do about the build error. It worked fine before I merged with main.
All it tells me is ```GENERATEPROJECTPRIFILE(0,0): Error : PRI175: 0x80073b0f - Processing Resources failed with error: Duplicate Entry. ```. 
I've tried turning up verbose logging to the "diagnostic" level, but it still doesn't show anything about where this is happening.
I've tried manually looking through each of the resource files changed in the merge for duplicates.
If anyone can help me debug this, it would be greatly appreciated.

winston-de commented:
It looks like the build blows up (the error in the previous comment) when I remove the ".Text" from the string resources, though I'm not really sure why. The only option I see is to create new ones, but that would mean loosing all the current translations.

winston-de commented:
Yay it worked!

duke7553 commented:
@winston-de Great! Once this is finished, we can finally start on the CommandbarFlyout update for the newly unified flyout.

winston-de commented:
> @winston-de Great! Once this is finished, we can finally start on the CommandbarFlyout update for the newly unified flyout.

I did a little bit of testing, it should be a relatively simple change as well.

winston-de commented:
I've been noticing that there's been some issues with users not being able to find items that are hidden unless you shift+right click. What do you think of moving those items to the "more" menu when shift isn't held?

It would look something like this:
<img src="https://user-images.githubusercontent.com/59544401/112772985-d1cd2800-8fe8-11eb-930a-8b1580b343fc.png" Width="500"/>

winston-de commented:
Inadvertently, this also mostly fixed #4331. I'll fix the nav toolbar one as well and just bundle the full fix in this pr.

duke7553 commented on Files/BaseLayout.cs:
```diff
@@ -475,6 +424,11 @@ protected override async void OnNavigatedTo(NavigationEventArgs eventArgs)
             catch (Exception e)
             {
             }
+
+            ItemContextMenuFlyout.Opening += ItemContextFlyout_Opening;
+            ItemContextMenuFlyout.AreOpenCloseAnimationsEnabled = AppSettings.AreRightClickContentMenuAnimationsEnabled;
+            BaseContextMenuFlyout.Opening += BaseContextFlyout_Opening;
+            BaseContextMenuFlyout.AreOpenCloseAnimationsEnabled = AppSettings.AreRightClickContentMenuAnimationsEnabled;
```
Should these events be unsubscribed in the OnNavigatingFrom() override?

duke7553 reviewed (dismissed):
LGTM

winston-de commented:
I think we should wait for #4218 to be merged before merging this one, as it adds a command to pin and unpin items to the Start menu. Currently pin to start doesn't do anything since there isn't a command.

yaichenbaum commented:
> @yaichenbaum I've updated all the localization keys (removed the ".Text"). Should I run the multilingual app toolkit?

@winston-de Sure!

winston-de commented:
> > @yaichenbaum I've updated all the localization keys (removed the ".Text"). Should I run the multilingual app toolkit?
> 
> @winston-de Sure! By the way, why do a lot of the keys have the number 2 attached to them?

@yaichenbaum I don't know why, but it wouldn't build unless I added them.

yaichenbaum commented:
@winston-de Have you checked to make sure there are no duplicate keys?

winston-de commented:
@yaichenbaum Yes, there weren't any. I think the problem initially was that I was not changing all the keys, just then ones in the en-us file, so the compiler saw that as duplicate keys. Now that I'm changing them all, I can probably remove the 2s without any issue, if you'd like.

winston-de commented:
Ok, so I tried removing the 2 for one set of resources, and it built just fine. Then I removed them all, and the build failed with the duplicate error. I checked each resource key for duplicates, and didn't find any. I'm guessing one of them is causing the problem for whatever, but I have no clue which one (except that it's not BaseLayoutContextFlyoutLayoutMode).

yaichenbaum commented:
@winston-de Another possibility is that one of the resources are being used where they shouldn't and removing the `.Text` is throwing the error. The best solution I can think of for now is to go through each string one at a time.

winston-de commented:
@yaichenbaum That would take me hours to do, are the 2s really that bad? It works fine as it is. And if so, is it possible that I could come back and fix them in their own PR?

winston-de commented:
Also, @TyJOrtiz is it alright with you that I make the necessary changes to implement the new context menus in Column View?

TyJOrtiz commented:
@winston-de Yes that's fine

yaichenbaum commented:
> @yaichenbaum That would take me hours to do, are the 2s really that bad? It works fine as it is. And if so, is it possible that I could come back and fix them in their own PR?

I don't want to cause all that extra work, how about keeping the new keys, but can you change the names of the new keys to something like `RegularContextFlyout...`? This will make it easier to search for the older keys and remove them in a future PR.

winston-de commented:
@yaichenbaum I can do that, but I don't see why you would to search through and remove the old keys, as they're gone. I changed the keys with find/replace throughout the entire solution so that the current translations wouldn't be lost.

winston-de commented:
The new context menus in Column View are up and running. 
![image](https://user-images.githubusercontent.com/59544401/113067967-32d43780-9172-11eb-884c-491f4b4c42f7.png)

Btw, @TyJOrtiz the column view is awesome! :)

yaichenbaum commented:
> @yaichenbaum I can do that, but I don't see why you would to search through and remove the old keys, as they're gone. I changed the keys with find/replace throughout the entire solution so that the current translations wouldn't be lost.

There is a duplicate key somewhere, I will try to find it.

yaichenbaum commented:
@winston-de It seems like the sort by "Date deleted" option is showing in all directories, additionally, the layout menu items seem to be disabled.

winston-de commented:
> > @yaichenbaum I can do that, but I don't see why you would to search through and remove the old keys, as they're gone. I changed the keys with find/replace throughout the entire solution so that the current translations wouldn't be lost.
> 
> There is a duplicate key somewhere, I will try to find it.

@yaichenbaum Thanks for doing this.
I searched every key for anything with the same name throughout the entire solution, and each one had somewhere between 37-54 references (which corresponds to the number of languages it was translated to). If there is a duplicate, I think it exists in only one resource file.

winston-de commented:
@yaichenbaum It looks like I may have never had to remove the ".Text" from everything. 😆 
I didn't realize it was possible to access them by using a slash, like this ```"resourcename/Text".GetLocalized()```.

winston-de commented:
@yaichenbaum I apologize for this, I just learned that the resources could be accessed like that.
